nuxeo-java-client/src/main/java/org/nuxeo/client/spi/NuxeoClientRemoteException.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (C) Copyright 2016-2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Vladimir Pasquier <vpasquier@nuxeo.com>
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.client.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Exception thrown by the client when a HTTP error occurred.
+ *
+ * @since 3.0
+ */
+public class NuxeoClientRemoteException extends NuxeoClientException {
```

Comment:
- I'd prefer `NuxeoHttpException`, which is shorter.
- I notice that there're no setters in your class: do you want to ensure the immutability of this class? If yes, the immutability is broken because of the protected fields, you need to change them to `private`.